components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/BasicClientAuthenticationHandler.java

```diff
@@ -137,7 +137,7 @@ protected AuthenticationResult doAuthenticate(MessageContext messageContext) thr
                         throw new AuthenticationFailException(errorMessage, e);
                     } catch (InvalidOAuthClientException | IdentityOAuth2Exception e) {
                         String errorMessage = "Invalid client: " + clientId;
-                        log.error(errorMessage, e);
+                        log.debug(errorMessage);
```

Comment:
Add `log.isDebugEnabled()` check

Reply:
We only add debug enabled check if there are expensive operations such as string concat. Since we don't have any here, hope this is fine.